src/cryptoadvance/specter/logic.py

```diff
@@ -751,8 +756,32 @@ def txonaddr(self, addr):
         txlist = [tx for tx in self.transactions if tx["address"] == addr]
         return len(txlist)
 
+    def balanceonaddr(self, addr):
+        balancelist = [utxo["amount"] for utxo in self.utxo if utxo["address"] == addr]
+        return sum(balancelist)
+
+    def txonlabel(self, label):
+        txlist = [tx for tx in self.transactions if (tx["label"] == label if "label" in tx else tx["address"] == label)]
+        return len(txlist)
+
+    def balanceonlabel(self, label):
+        balancelist = [utxo["amount"] for utxo in self.utxo if utxo["label"] == label or utxo["address"] == label]
```

Comment:
please add a check that utxo has "label" attribute. Crashes if my change address doesn't have a label.